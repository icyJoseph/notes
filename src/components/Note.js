import React from "react";

const Note = ({ note }) => {
  return <li className="note">{note}</li>;
};

Note.displayName = "Note";

export default Note;
