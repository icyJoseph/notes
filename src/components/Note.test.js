import React from "react";
import { shallow } from "enzyme";

import Note from "./Note";

describe("Note", () => {
  const props = { note: "Use me to test" };
  const note = shallow(<Note {...props} />);

  it("renders properly", () => {
    expect(note).toMatchSnapshot();
  });

  it("displays the received props", () => {
    expect(note.find(".note").text()).toEqual(props.note);
  });
});
