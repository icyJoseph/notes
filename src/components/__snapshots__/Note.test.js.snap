// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    note="Use me to test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Use me to test",
      "className": "note",
    },
    "ref": null,
    "rendered": "Use me to test",
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Use me to test",
        "className": "note",
      },
      "ref": null,
      "rendered": "Use me to test",
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
